src/Nox.Integration.SqlServer/Extensions/SqlServerIntegrationOptionsExtension.cs

```diff
@@ -0,0 +1,24 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Nox.Abstractions;
+using Nox.EntityFramework.SqlServer;
+using Nox.Infrastructure;
+using Nox.Solution;
+using Nox.Types.EntityFramework.Abstractions;
+using Nox.Types.EntityFramework.Enums;
+
+namespace Nox.Integration.SqlServer;
+
+public static class SqlServerIntegrationOptionsExtension
+{
+    public static NoxIntegrationOptionsBuilder WithSqlServerStore(this NoxIntegrationOptionsBuilder optionsBuilder)
+    {
+        optionsBuilder.Services.AddSingleton<INoxDatabaseProvider>(provider => new SqlServerDatabaseProvider(
```

Comment:
like we agree lets have a `IEntityDatabaseProvider: INoxDatabaseProvider`

and `IIntegrationDatabaseProvider: INoxdatabaseProvider` for example